### ISSUE TITLE
`gulp` as a `peerDependency`

### ISSUE THREAD

**Issue by destroyerofbuilds** [mlucool/gulp-jsdoc3#10]
On January 9th, eed146deb950484e1ccdf85c5630f31998c4826d was pushed to this repository.

In that commit `gulp` was moved from a `devDependency` to a `peerDependency`. The behavior of `npm` version 2 is to install peer dependencies. So a project using `gulp`, along with this plugin, will get `gulp` `3.9.0` or higher (but major version `3`). 

Recently I started using `gulp` version 4 (It's still in beta as a `4` branch on the `gulp` repo). 

That requires that I list `gulp` 4 as a `devDependency` of my project. I also use this plugin in my project. This causes a version conflict between my project's gulp version, and the `peerDependency` of this project. Inevitably the `gulp` version required by this project gets installed causing my project's build to fail.

As this project has no strict dependency on `gulp`, other than that this plugin is used in a stream, could the `peerDependency` requirement be either removed, or relaxed to include `gulp` version `4`.

On a side note, this project works great with version `4` (Worked well until the aforementioned changed)

**Comment by mlucool** [comment]
I'll bump the compatible versions as soon as I test. ETA sometime in the next week or so. Thanks for the heads up and feedback!

**Comment by mlucool** [comment]
TLDR: After some testing and thought I am removing the gulp peerDependency.

It seems like a nightmare for alpha versions: http://blog.npmjs.org/post/98131109725/npm200  I agree there is no direct dependency on gulp here in the dist code, but I liked having peer to imply I won't deal with bugs in an older version of gulp. Still, I'd rather enable people to move to gulp4 easily so this has been removed: https://github.com/mlucool/gulp-jsdoc3/commit/906c1d33bf8d275c4b4ee6c25cd9cacdc88cabf9
